subsys/bluetooth/host/gatt.c

```diff
@@ -534,6 +534,32 @@ static const struct bt_gatt_attr *find_attr(uint16_t handle)
 	return attr;
 }
 
+static void gatt_insert(struct bt_gatt_service *svc, u16_t last_handle)
+{
+	struct bt_gatt_service *tmp, *prev = NULL;
+
+	if (svc->attrs[0]->handle > last_handle) {
+		sys_slist_append(&db, &svc->node);
+		return;
+	}
+
+	prev = NULL;
+
+	/* DB shall always have its service in ascending order */
+	SYS_SLIST_FOR_EACH_CONTAINER(&db, tmp, node) {
+		if (tmp->attrs[0]->handle > svc->attrs[0]->handle) {
```

Comment:
You should probably build check these since I didn't and apparently it is not building correcly e.g. it should have been attrs[].handle, etc.

Reply:
oh thank you, my bad. I forgot to compile with CONFIG_BT_GATT_DYNAMIC_DB=y :-(.
It's fixed now!